src/game_battlealgorithm.cpp

```diff
@@ -1232,24 +1232,24 @@ bool Game_BattleAlgorithm::Skill::Execute() {
 
 			effect = Utils::Clamp(effect, 0, MaxDamageValue());
 
-			if (skill.affect_hp) {
+			if (skill.affect_hp && Utils::PercentChance(to_hit)) {
 				if (Player::IsRPG2k3()) {
 					this->hp = effect;
 				} else {
 					this->hp = std::max<int>(0, std::min<int>(effect, GetTarget()->GetMaxHp() - GetTarget()->GetHp()));
 				}
 			}
-			if (skill.affect_sp) {
+			if (skill.affect_sp && Utils::PercentChance(to_hit)) {
 				int sp_cost = GetSource() == GetTarget() ? source->CalculateSkillCost(skill.ID) : 0;
 				this->sp = std::max<int>(0, std::min<int>(effect, GetTarget()->GetMaxSp() - GetTarget()->GetSp() + sp_cost));
 			}
-			if (skill.affect_attack)
+			if (skill.affect_attack && Utils::PercentChance(to_hit))
 				this->attack = std::max<int>(0, std::min<int>(effect, std::min<int>(GetTarget()->MaxStatBattleValue(), GetTarget()->GetBaseAtk() * 2) - GetTarget()->GetAtk()));
-			if (skill.affect_defense)
+			if (skill.affect_defense && Utils::PercentChance(to_hit))
 				this->defense = std::max<int>(0, std::min<int>(effect, std::min<int>(GetTarget()->MaxStatBattleValue(), GetTarget()->GetBaseDef() * 2) - GetTarget()->GetDef()));
-			if (skill.affect_spirit)
+			if (skill.affect_spirit && Utils::PercentChance(to_hit))
 				this->spirit = std::max<int>(0, std::min<int>(effect, std::min<int>(GetTarget()->MaxStatBattleValue(), GetTarget()->GetBaseSpi() * 2) - GetTarget()->GetSpi()));
-			if (skill.affect_agility)
+			if (skill.affect_agility && Utils::PercentChance(to_hit))
```

Comment:
Confirmed the `to_hit` is re-reolled for every stat which is affected by the skill. Nice catch!